zcash_client_sqlite/src/lib.rs

```diff
@@ -402,6 +403,29 @@ impl<'a, P: consensus::Parameters> DataConnStmtCache<'a, P> {
 
 #[allow(deprecated)]
 impl<'a, P: consensus::Parameters> WalletWrite for DataConnStmtCache<'a, P> {
+    fn create_account(
+        &mut self,
+        seed: SecretVec<u8>,
+    ) -> Result<(AccountId, UnifiedSpendingKey), Self::Error> {
+        self.transactionally(|stmts| {
+            let account = wallet::get_max_account_id(stmts.wallet_db)?
+                .map(|a| AccountId::from(u32::from(a) + 1))
```

Comment:
In a pairing, @nuttycom and I confirmed that if `a` here is (the theoretical) `u31::MAX`, then this increment will be detected inside `UnifiedSpendingKey::from_seed` (by the `orchard` crate) and an error will be returned.

We separately identified that the Sapling ZIP 32 impl could really use some love.